sdk/eventgrid/Azure.Messaging.EventGrid/tests/TestEvent.Serialization.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.Json;
+using Azure.Core;
+
+namespace Azure.Messaging.EventGrid.Tests
+{
+    internal partial class TestEvent : IUtf8JsonSerializable
```

Comment:
Hmm, so customers wouldn't be able to do this because IUtf8JsonSerializable is internal.
It looks like we will need to do some magic to get this working in our convenience layer because the Track 2 code gen doesn't support serializing custom types. Let's leave off the custom schema test from this PR and I will create a separate issue for the work needed to support custom schema. It's fine to leave in the resource creation for custom schema, and you can leave in the test class, but we can drop this file.